src/gc.rs

```diff
@@ -21,15 +21,10 @@ type StackValue = usize;
 type StackScanCallback = extern "sysv64" fn(&mut Collector, StackAddress);
 #[link(name = "SpillRegisters", kind = "static")]
 extern "sysv64" {
-    // Rust (correctly) warns us that `Collector` has an unspecified layout and
-    // that it is UB to pass this to C code via FFI. However, our register spill
-    // code does not dereferences this, nor does it ever move the `Collector`
-    // instance by value. It simply moves the address to the first argument
-    // register to be used as the implicit `self` address in the callback. It is
-    // therefore safe to do this without specifying a C style layout for the
-    // struct.
-    #[allow(improper_ctypes)]
-    fn spill_registers(collector: &mut Collector, callback: StackScanCallback);
+    // Pass a type-punned pointer to the collector and move it to the asm spill
+    // code. This is so it can be passed straight back as the implicit `self`
+    // address in the callback.
+    fn spill_registers(collector: *mut i8, callback: StackScanCallback);
```

Comment:
I think it's more conventional to use `*mut u8`, but I might be wrong.

Reply:
Sure, I think I probably meant to write `u8` anyway.